apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/ListFeatureAssertionsSpec.kt

```diff
@@ -0,0 +1,33 @@
+package ch.tutteli.atrium.api.infix.en_GB
+
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.specs.feature1
+import ch.tutteli.atrium.specs.fun2
+import ch.tutteli.atrium.specs.notImplemented
+
+class ListFeatureAssertionsSpec : ch.tutteli.atrium.specs.integration.ListFeatureAssertionsSpec(
+    feature1<List<Int>, Int, Int>(Expect<List<Int>>::get),
+    fun2<List<Int>, Int, Expect<Int>.() -> Unit>(Expect<List<Int>>::get),
+    feature1<List<Int?>, Int, Int?>(Expect<List<Int?>>::get),
+    fun2<List<Int?>, Int, Expect<Int?>.() -> Unit>(Expect<List<Int?>>::get)
+) {
+
+    companion object {
+
+        @Suppress("unused", "UNUSED_VALUE")
+        private fun ambiguityTest() {
+            var a1: Expect<List<Int>> = notImplemented()
+            var a2: Expect<out List<Int>> = notImplemented()
+
+            getPlant(a1, 1)
+            getPlant(a1, Index(1))
+            a1 = a1.get(1) { }
+            getPlant(a2, 1)
+            getPlant(a2, Index(1))
+            a2 = a2.get(1) { }
+        }
+
+        private fun getPlant(plant: Expect<out List<Int>>, index: Int) = plant get index
+        private fun getPlant(plant: Expect<out List<Int>>, index: Index) = plant get index
```

Comment:
The ambiguity test shall reveal bugs in the implementation such as `Expect<Iterable<*>>.hasNext()` does not work. However, for this we need to duplicate code, therefore, please remove this function. Please rebase on master, I have just improved the ambiguityTests in atrium-api-fluent-en_GB. Copy the newer version.

Reply:
Not sure what function exactly you ask to remove.
In issue description there was a point, where you asked to check "real" infix function usage and this `getPlant` function is doing it for different indices.
You want me to remove both?
How ambiguity test will look like for me then? Is it something like this then?
```
        a1.get(1)
        a2.get(1)
        a1b.get(1)
        a2b.get(1)
        a3.get(1)
```

Reply:
Both getPlant functions. Use the you API defined directly. Use tge overload which expects `Index` for the case with the assertionCreator